### ISSUE TITLE
Mica material

### ISSUE THREAD

**Issue by niels9001** [microsoft/microsoft-ui-xaml#5265]
Windows 11 introduces a new material Mica, similar to Acrylic. Some questions about this new material:

- When will it be available for WinUI 2.x and 3?
- Is this material back-ported to Windows 10 as well?

**Comment by chingucoding** [comment]
> When will it be available for WinUI 2.x and 3?

It already is in WinUI 2.6.

> Is this material back-ported to Windows 10 as well?

Afaik, no.

**Comment by niels9001** [comment]
@chingucoding... Do you have link to docs?

And what does it fall back to on Windows 10? And.. when can we expect it to ship with WinUI 3.x - since Acrlyic isn't working either.

**Comment by chingucoding** [comment]
Regarding the back-port not sure where I picked it up, regarding WinUI 2.6, it was announced here: https://github.com/microsoft/microsoft-ui-xaml/issues/5262

**Comment by mdtauk** [comment]
Mica does seem to fulfill my asks for #921 #920 issues, and my constant ask about a Faux Acrylic.

It is worded as a WinUI 2.6 feature, so surely it gets backported to Windows 10?

And don't think I'm not already expecting the day when Host Backdrop Acrylic is deemed too hard to include in WinUI 3, and the official decision is that **Mica** replaces that usage lol

**Comment by jevansaks** [comment]
> Mica does seem to fulfill my asks for #921 #920 issues, and my constant ask about a Faux Acrylic.

Yes, it's been hard not to tell you about this for a long time. :)

> so surely it gets backported to Windows 10?

We do not plan to backport this to Windows 10. It takes advantage of a new API in Windows11 where we can get a blurred version of the wallpaper (just wallpaper, not the windows behind). So for now Mica falls back to solid color downlevel, and it is our recommended material for persistent windows.

The BackdropMaterial API is in WinUI 2.6 so that your app can always use it and not have to do version checks, but under the covers it decides what to do based on OS capabilities and renders Mica or fallback color.

One possibility we could consider is to plug in the user accent color into Mica downlevel -- it would give a similar feel of a touch of color for the active window instead of being completely neutral. 

> when can we expect it to ship with WinUI 3.x - since Acrlyic isn't working either.

When WinUI 2.6 is ported to WinUI 3.x, Mica/BackdropMaterial will be included. That's in plan to do for Reunion 1.0 but I don't know exactly when.

**Comment by mdtauk** [comment]
> > Mica does seem to fulfill my asks for #921 #920 issues, and my constant ask about a Faux Acrylic.
> 
> Yes, it's been hard not to tell you about this for a long time. :)
> 
> > so surely it gets backported to Windows 10?
> 
> We do not plan to backport this to Windows 10. It takes advantage of a new API in Windows11 where we can get a blurred version of the wallpaper (just wallpaper, not the windows behind). So for now Mica falls back to solid color downlevel, and it is our recommended material for persistent windows.
> 
> The BackdropMaterial API is in WinUI 2.6 so that your app can always use it and not have to do version checks, but under the covers it decides what to do based on OS capabilities and renders Mica or fallback color.
> 
> One possibility we could consider is to plug in the user accent color into Mica downlevel -- it would give a similar feel of a touch of color for the active window instead of being completely neutral. 
> 
> > when can we expect it to ship with WinUI 3.x - since Acrlyic isn't working either.
> 
> When WinUI 2.6 is ported to WinUI 3.x, Mica/BackdropMaterial will be included. That's in plan to do for Reunion 1.0 but I don't know exactly when.
> 
> 

@jevansaks a couple of thoughts.  Mica takes the wallpaper and uses that to display the brush.

Now I seem to remember all the way back in Windows 7 and 8 the ability for the Aero glass to grab a predominant colour from your wallpaper.

Does that API still exist in DWM?  could that be used by *Mica* to grab the average or predominant colour from your wallpaper, do the light and dark colour math, and give us a solid colour that is tinted?

Of course the colour would still shift towards neutral for inactive windows like at present.

Using the accent colour would lead to a more monotone app experience I fear.

**Comment by jevansaks** [comment]
> Now I seem to remember all the way back in Windows 7 and 8 the ability for the Aero glass to grab a predominant colour from your wallpaper.
> 
> Does that API still exist in DWM? could that be used by _Mica_ to grab the average or predominant colour from your wallpaper, do the light and dark colour math, and give us a solid colour that is tinted?

It was never an API, it was (and still is, I think?) a behavior in Settings you can choose to have your accent color based off your wallpaper.

Unfortunately the whole reason we need this new API is that low-IL apps cannot read the user's wallpaper (including things like analyzing it for color palette). So the only personalization color we have available as API is the accent color. 

> Using the accent colour would lead to a more monotone app experience I fear.

Right, that's partly why we didn't pursue it.

**Comment by mdtauk** [comment]
> > Now I seem to remember all the way back in Windows 7 and 8 the ability for the Aero glass to grab a predominant colour from your wallpaper.
> > Does that API still exist in DWM? could that be used by _Mica_ to grab the average or predominant colour from your wallpaper, do the light and dark colour math, and give us a solid colour that is tinted?
> 
> It was never an API, it was (and still is, I think?) a behavior in Settings you can choose to have your accent color based off your wallpaper.
> 
> Unfortunately the whole reason we need this new API is that low-IL apps cannot read the user's wallpaper (including things like analyzing it for color palette). So the only personalization color we have available as API is the accent color.
> 
> > Using the accent colour would lead to a more monotone app experience I fear.
> 
> Right, that's partly why we didn't pursue it.

That is unfortunate, I thought if not a public API it may be a registry value that could be read.

Don't suppose its something the Windows team could through into a cumulative update lol 😁

**Comment by mdtauk** [comment]
Thinking on it, Background Acrylic with a stronger tint colour could be used as a fall back on Windows 10.

**Comment by massijay** [comment]
> Does that API still exist in DWM? could that be used by _Mica_ to grab the average or predominant colour from your wallpaper, do the light and dark colour math, and give us a solid colour that is tinted?

I was thinking exactly about this, it would be really nice to have a fallback solid color like this. If it's not already available as an API you should consider doing it :) (maybe also backport it to Windows 10)

What about the ability to set a custom image as a source for the Mica material? Are you considering this?

**Comment by jevansaks** [comment]
> Thinking on it, Background Acrylic with a stronger tint colour could be used as a fall back on Windows 10.

@mdtauk Most windows don't have enough color to have the "content behind my window" brush achieve the same goal as pulling from the wallpaper. Also the content behind the window isn't a user personalization choice, it's just some random stuff. 

> What about the ability to set a custom image as a source for the Mica material? Are you considering this?

@massijay That's an interesting idea, but that would be more to help the app's brand colors come through as opposed to the user's colors.

**Comment by mdtauk** [comment]
I think the idea of providing a Fallback Image Source would be a neat idea.  Maybe WinUI could also bundle one in the same way it bundles the Noise Asset

**Comment by massijay** [comment]
> @massijay That's an interesting idea, but that would be more to help the app's brand colors come through as opposed to the user's colors.

Exactly, like the Accent color I think that also the Mica material should have the possibility to reflect the brand's colors.

> I think the idea of providing a Fallback Image Source would be a neat idea. Maybe WinUI could also bundle one in the same way it bundles the Noise Asset

I was thinking more like an override of the source image rather than a fallback one, but one thing doesn't exclude the other, having both options (ovverride and fallback) would be even better
